src/components/expression/expression-renderer.js

```diff
@@ -518,34 +704,45 @@ genotet.ExpressionRenderer.prototype.drawGeneProfiles_ = function() {
     })
     .y(yScale)
     .interpolate('linear');
+
   this.data.profiles.forEach(function(profile, i) {
-    var geneIndex = this.data.profiles[i].row;
-    var pathColor = this.COLOR_CATEGORY_[
-    genotet.utils.hashString(
-      heatmapData.geneNames[geneIndex]
+    var pathColor = this.COLOR_CATEGORY_[genotet.utils.hashString(
+      heatmapData.geneNames[profile.row]
     ) % this.COLOR_CATEGORY_SIZE_];
-    this.data.profiles[i] = {
-      geneName: heatmapData.geneNames[geneIndex],
-      row: geneIndex,
-      color: pathColor
-    };
+    this.data.profiles[i].color = pathColor;
+    var legendWidth = i * (legendHeight +
+      profile.geneName.length * this.GENE_LABEL_WIDTH_FACTOR_);
+    legend.append('rect')
+      .attr('height', legendHeight)
+      .attr('width', legendHeight)
+      .attr('x', legendWidth)
+      .style('fill', pathColor);
+    legend.append('text')
+      .text(profile.geneName)
+      .attr('transform', genotet.utils.getTransform([
+        legendHeight + 1,
```

Comment:
Please explain the "+1" by either comments or using a self-explained constant name.